### PR TITLE
feat: add `Ord` and deriving instance for it.

### PR DESCRIPTION
**Pull request by DanielFabian**
For many data structures having an ordering is necessary. This one adds the `Ord` type class and a deriving handler for it. The ordering is based on order of constructors followed by lexicographical ordering within a constructor.

edit: now with the inhabited instance for `Ordering`, it compiles fine (though it, too, uses `partial`, internally)

### CODE REVIEW

File: src/Init/Data/Ord.lean [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Author: Dany Fabian
+-/
+
+prelude
+import Init.Data.Int
+import Init.Data.String
+
+inductive Ordering
+| LT | EQ | GT
+deriving Inhabited
```
Comment (leodemoura) [inline src/Init/Data/Ord.lean:13]: We do not capitalize constructors.

File: src/Init/Data/Ord.lean [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class Ord (α : Type u) where
+  compare : α → α → Ordering
+
+def cmp [inst : Ord α] := inst.compare
+
+def compareOfLessAndEq {α} (x y : α) [HasLess α] [Decidable (x < y)] [DecidableEq α] :=
+    if x < y then Ordering.LT
```
Comment (leodemoura) [inline src/Init/Data/Ord.lean:22]: We use 2 spaces for indentation.

**Comment by leodemoura** [comment]
Thanks. It looks good. I have only two minor comments.

File: src/Init/Data/Ord.lean (hunk 1 above)
Comment (Kha) [inline src/Init/Data/Ord.lean:13]:
```suggestion
inductive Ordering where
  | lt | eq | gt
  deriving Inhabited
```

File: src/Init/Data/Ord.lean [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+| LT | EQ | GT
+deriving Inhabited
+
+
+class Ord (α : Type u) where
+  compare : α → α → Ordering
+
+def cmp [inst : Ord α] := inst.compare
+
+def compareOfLessAndEq {α} (x y : α) [HasLess α] [Decidable (x < y)] [DecidableEq α] :=
```
Comment (Kha) [inline src/Init/Data/Ord.lean:21]: Missing result type

File: src/Init/Data/Ord.lean [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    else if x = y then Ordering.EQ
+    else Ordering.GT
+
+instance : Ord Nat where
+  compare x y := compareOfLessAndEq x y
+
+instance : Ord Int where
+  compare x y := compareOfLessAndEq x y
+
+instance : Ord Int where
```
Comment (Kha) [inline src/Init/Data/Ord.lean:32]: duplicate

File: src/Init/Data/Ord.lean [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+inductive Ordering where
+  | lt | eq | gt
+deriving Inhabited
+
+
+class Ord (α : Type u) where
+  compare : α → α → Ordering
+
+def cmp [inst : Ord α] := inst.compare
```
Comment (gebner) [inline src/Init/Data/Ord.lean:19]:
```suggestion
export Ord (compare)
```

Comment (DanielFabian) [reply src/Init/Data/Ord.lean:19]: i see, ok, then I'll have to change the code gen too. Will do.

Comment (DanielFabian) [reply src/Init/Data/Ord.lean]: done

**Pull request merged**